docs/tutorials/writing_docs.rst

```diff
@@ -0,0 +1,244 @@
+- Better example: "Writing a result parser module". That module.
+- Best example: "Result parser module" within a "Plugin Files" section in a
+  tutorial document about writing result parser plugins. We
+  know by context this a plugin file and that we'll be writing it.
+
+Documentation Type
+``````````````````
+There are four types of documentation within Pavilion:
+
+Usage
+  The purpose of usage documentation is to give the reader a quick
+  introduction to the overall structure of Pavilion. It should not try to
+  completely document any subject, and should instead link to the **Feature**
+  documentation on that topic. It may be mildly conversational in tone. Most
+  directories have a "basics" document that is of this type.
+Tutorial
+  Tutorials are meant to guide the user through the process of using the
+  software or adding to it. They may be mildly conversational in tone. They
+  should be concise but *complete*. They should cover each step in the
+  process, why that step exists, what happens when things go right, and
+  how to resolve issues when things go wrong.
+Feature
+  Feature documentation should attempt to completely detail how to use each
+  feature of the software. It may, when necessary, cover limits or other
+  considerations imposed by implementation details. It should be as concise
+  as possible, and should not be conversational in tone.
+API
+  API documentation is primarily written as part of the code. It is organized
+  by hand in documents under ``docs/source``, but the content is generated
+  from doc-strings using Sphinx's autodoc feature. Every source file,
+  class, method and function should be documented here. Function/methods
+  should have their arguments, return values, and exceptions raised documented.
+  Public globals, class and instance variables should also be documented.
+
+Audience
+````````
+The audience for Pavilion and its documentation is a technical one. You can
+generally assume the your audience understands at least the following:
+
+- Basic computer science concepts like regular expressions.
+- Shell scripting.
+- YAML
+- Cluster computing and schedulers
+- (Plugin Tutorials) Python basics including classes
+
+
+.. _demonstrative:
+
+Being Demonstrative
+-------------------
+
+It is often much faster to show than to tell, and readers will frequently
+read nothing but your example code and infer the rest. Such code should be
+well-documented and broad, condensing everything needed into a few
+comprehensive examples.
+
+- Examples should be highlighted according the the language they're written in.
+
+  - Sphinx requires that such code be a valid snippet for that language.
+- You don't need to introduce examples. Simply end the last sentence of the
+  preceding paragraph with a colon ``":"``.
+- Most lines should have a explanatory comment.
+- Where ellipsis would be used to denote that the example is an excerpt,
+  a comment should state that instead.
```

Comment:
Would this necessarily be true if, for example, you're trying to show a few key elements of a JSON dictionary?  I tend to use ellipses in such situations.